src/activities/bargame/bargame.js

```diff
@@ -0,0 +1,196 @@
+    items.boxes.columns = boardSize[sublevel-1];
+
+    // Mask setup refresh
+    items.masks.columns = boardSize[sublevel-1];
+
+    // Hiding all visible balls
+    for (var x = 0; x < items.answerBallsPlacement.columns; x++) {
+        items.answerBallsPlacement.children[x].opacity = 0.0;
+    }
+    items.answerBallsPlacement.columns = boardSize[sublevel-1]
+
+    // Resetting ball plate
+    items.rootWindow.source = url + "school_bg" + level + ".svg"
+
+}
+
+function nextLevel() {
+    level++;
+    if (level > 4) {
+        level = 1;
+    }
+    initLevel();
+}
+
+function previousLevel() {
+    level--;
+    if (level < 1) {
+        level = 4;
+    }
+    initLevel();
+}
+
+function calculateWinPlaces() {
+    /* Calculates all the possible winning moves in the
+    available board size. It generates a list all winning
+    moves for the computer */
+    var winners = [];
+    var winnersList = [];
+    var min = numberBalls[sublevel - 1][0];
+    var max = numberBalls[sublevel - 1][1];
+    var period = (min + max);
+
+    for (var x = 0; x < min; x++) {
+        winnersList.push((boardSize[sublevel - 1] - 1 - x) % period);
+    }
+
+    for (var x = 0; x < (boardSize[sublevel - 1]); x++ ) {
+        if (winnersList.indexOf((x + 1) % period) >= 0) {
+            winners.push(x);
+        }
+    }
+
+    var level_win = (level - 1) * min;
+
+    if (level_win == 0) {
+        winners = [];
+    } else {
+        winners = winners.slice(-level_win);
+    }
+
+    listWin =  winners;
+}
+
+function machinePlay() {
+    function accessible(x) {
+        if (listWin.indexOf(x + moveCount) >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function randomNumber(minimum, maximum){
+        return Math.round( Math.random() * (maximum - minimum) + minimum);
+    }
+
+    var playable = [];
+
+    for (var x = numberBalls[sublevel - 1][0]; x < numberBalls[sublevel - 1][1] + 1; x++) {
+        if (accessible(x)) {
+            playable.push(x);
+        }
+    }
+    if (playable.length != 0) {
+        var value = playable[Math.floor(Math.random()*playable.length)];
+    } else {
+        var value = randomNumber(numberBalls[sublevel - 1][0], numberBalls[sublevel - 1][1]);
+    }
+
+    play(2, value);
+}
+
+function play(player, value) {
+    for (var x = 0; x < value ; x++) {
+        moveCount++;
+        if (moveCount <= (boardSize[sublevel - 1] - 1)) {
+            if (player == 1) {
+                items.answerBallsPlacement.children[moveCount].opacity = 1.0;
+                items.answerBallsPlacement.children[moveCount].source = url + "green_ball.svg";
+            } else {
+                items.answerBallsPlacement.children[moveCount].opacity = 1.0;
+                items.answerBallsPlacement.children[moveCount].source = url + "blue_ball.svg";
+            }
+        }
+        if (moveCount == (boardSize[sublevel - 1] - 1)) {
+            // Level increment
+            level++;
```

Comment:
I would not change level when winning, in this kind of games, the child may want to try multiple time as same level to find new ways of playing